virtool/administrators/data.py

```diff
@@ -177,3 +177,12 @@ async def delete(self, user_id: str):
             return
 
         raise ResourceNotFoundError()
+
+    async def promote_administrators(self):
+        async with self._db.create_session() as session:
```

Comment:
Session doesn't add anything here. You are not modifying MongoDB, which is where as transaction is important.